platform/default/src/mbgl/storage/pmtiles_file_source.cpp

```diff
@@ -0,0 +1,547 @@
+#include <sstream>
+#include <map>
+
+#include <mbgl/platform/settings.hpp>
+#include <mbgl/storage/pmtiles_file_source.hpp>
+#include <mbgl/storage/file_source_manager.hpp>
+#include <mbgl/storage/file_source_request.hpp>
+
+#include <rapidjson/document.h>
+#include <rapidjson/rapidjson.h>
+#include <rapidjson/stringbuffer.h>
+#include <rapidjson/writer.h>
+
+#include <mbgl/util/thread.hpp>
+#include <mbgl/util/url.hpp>
+#include <mbgl/util/chrono.hpp>
+#include <mbgl/util/compression.hpp>
+#include <mbgl/util/filesystem.hpp>
+
+#include <pmtiles.hpp>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+
+#if defined(__QT__) && (defined(_WIN32) || defined(__EMSCRIPTEN__))
+#include <QtZlib/zlib.h>
+#else
+#include <zlib.h>
+#endif
+
+namespace {
+// https://github.com/protomaps/PMTiles/blob/main/spec/v3/spec.md#3-header
+constexpr int PMTILES_HEADER_OFFSET = 0;
+constexpr int PMTILES_HEADER_LENGTH = 127;
```

Comment:
I think all upper case names should be reserved for macros.

Reply:
Originally I wanted to declare these numbers as defines, but how to use them correctly? The compiler complains when I define them using #define.

Reply:
Keep it `constexpr`, but lowercase them.

Also camelcase maybe `pmtilesHeaderOffset`?